bin/wasm-node/javascript/src/index.js

```diff
@@ -49,24 +63,55 @@ export async function start(config) {
   const worker = new Worker(new URL('./worker.js', import.meta.url));
   let workerError = null;
 
+  // map of chain ids to health requests - their ids and the their promise resolve and reject funtions
+  // { 1: [{ requestId: <n>, resolve: fn(), reject: fn() }, /* ... */ }]
+  const chainHealthRequestCallbacks =  {};
```

Comment:
This doesn't need to track the `chainId`. It can just directly track the values generated from `nextHealthRequestId`.

Reply:
You need to track the `chainId` internally, though, and clean up entries when `remove()` is called on a chain.

Reply:
Yeah that also makes the code much neater

Reply:
Done